cmd/data-usage.go

```diff
@@ -212,9 +285,17 @@ func (f *folderScanner) scanQueuedLevels(ctx context.Context, folders []cachedFo
 		}
 		thisHash := hashPath(folder.name)
 
-		if _, ok := f.oldCache.Cache[thisHash]; f.withFilter != nil && ok {
+		// If there are lifecycle rules for the prefix, remove the filter.
+		filter := f.withFilter
+		if f.oldCache.Info.lifeCycle != nil && filter != nil {
+			if f.oldCache.Info.lifeCycle.HasActiveRules(folder.name) {
```

Comment:
I see that you are passing folder.name here, which seems to contain bucket/prefix, However, HasActiveRules should only have prefix in it, without bucket name.

(Lifecycle is per bucket, and Lifecycle.Rule[].Filter.Prefix contains only the prefix without a bucket name in it)